src/common/gui/WhitelabelThemeGenerator.ts

```diff
@@ -0,0 +1,72 @@
+import { BaseThemeId, Theme } from "./theme"
+
+/**
+ * Whitelabel theme customizations that can be set
+ */
+export interface WhitelabelThemeSettings {
+	/**
+	 * Primary color in hexadecimal
+	 */
+	accentColor: string
+
+	/**
+	 * Theme to use
+	 */
+	theme: BaseThemeId
+
+	/**
+	 * Optional logo (if unspecified, the app's respective default grayscale logo for that theme will be used)
+	 */
+	logo?: string
+}
+
+export class WhitelabelThemeGenerator {
```

Comment:
does it make sense to make it a class? It doesn't have any state or dependencies and I don't think it will gain any

Reply:
Or does it aid with testing a lot?